File: MvvmSample/app/src/main/java/com/abben/mvvmsample/ui/movies/adapter/MoviesRecycleViewAdapter.java
package com.abben.mvvmsample.ui.movies.adapter;

import android.content.Context;
import android.databinding.DataBindingUtil;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.LinearLayout;

import com.abben.mvvmsample.R;
import com.abben.mvvmsample.bean.Movie;
import com.abben.mvvmsample.common.OnItemClickListener;
import com.abben.mvvmsample.databinding.ItemRecycleviewBinding;
import com.bumptech.glide.Glide;

import java.util.List;

import static com.bumptech.glide.request.RequestOptions.fitCenterTransform;

/**
 * Created by abben on 2017/5/3.
 */
public class MoviesRecycleViewAdapter extends RecyclerView.Adapter<MoviesRecycleViewAdapter.CustomVH> {

    private List<Movie> movies;
    private Context context;
    private int imageViewWidth;
    private OnItemClickListener<Movie> mOnItemClickListener;

    public MoviesRecycleViewAdapter(Context context, int screenWidth) {
        this.context = context;
        this.imageViewWidth = screenWidth / 2 - 20;
    }

    @Override
    public CustomVH onCreateViewHolder(ViewGroup parent, int viewType) {
        ItemRecycleviewBinding itemRecycleviewBinding = DataBindingUtil
                .inflate(LayoutInflater.from(parent.getContext()), R.layout.item_recycleview, parent, false);
        return new CustomVH(itemRecycleviewBinding);
    }

    public void setMovies(List<Movie> movies) {
        this.movies = movies;
        notifyDataSetChanged();
    }

    public void setOnItemClikeListen(OnItemClickListener<Movie> onItemClikeListen) {
        this.mOnItemClickListener = onItemClikeListen;
    }

    @Override
    public void onBindViewHolder(final CustomVH holder, int position) {
        if (mOnItemClickListener != null) {
            holder.itemView.setOnClickListener(view ->
                    mOnItemClickListener.onItemClick(movies.get(holder.getLayoutPosition()), view)
            );
        }

        holder.binding.setMovie(movies.get(position));

        ImageView myImageView = holder.binding.movieCover;
        int imageViewHeight = imageViewWidth * Integer.parseInt(movies.get(position).getImageOfMovieHeight()) /
                Integer.parseInt(movies.get(position).getImageOfMovieWidth());
        LinearLayout.LayoutParams imageViewParams = new LinearLayout.LayoutParams(imageViewWidth, imageViewHeight);
        final int imagePaddingLeftOrRight = 10;
        myImageView.setPadding(imagePaddingLeftOrRight, 10, imagePaddingLeftOrRight, 10);
        myImageView.setLayoutParams(imageViewParams);

        Glide.with(context)
                .load(movies.get(position).getImageOfMovie())
                .apply(fitCenterTransform())
                .into(myImageView);

        holder.binding.executePendingBindings();
    }

    @Override
    public int getItemCount() {
        return movies == null ? 0 : movies.size();
    }

    class CustomVH extends RecyclerView.ViewHolder {

        private ItemRecycleviewBinding binding;

        private CustomVH(ItemRecycleviewBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }

    }
}

File: MvvmSampleJetpack/app/src/main/java/com/abben/mvvmsamplejetpack/ui/movies/adapter/MoviesRecycleViewAdapter.kt
package com.abben.mvvmsamplejetpack.ui.movies.adapter

import android.view.LayoutInflater
import android.view.ViewGroup
import android.widget.LinearLayout
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.RecyclerView
import com.abben.mvvmsamplejetpack.R
import com.abben.mvvmsamplejetpack.bean.Movie
import com.abben.mvvmsamplejetpack.common.OnItemClickListener
import com.abben.mvvmsamplejetpack.databinding.ItemRecycleviewBinding
import com.bumptech.glide.Glide
import com.bumptech.glide.request.RequestOptions.fitCenterTransform

/**
 *  @author abben
 *  on 2019/1/21
 */
class MoviesRecycleViewAdapter(var imageViewWidth: Int) : RecyclerView.Adapter<MoviesRecycleViewAdapter.CustomVH>() {

    private var movies: List<Movie>? = null
    var mOnItemClickListener: OnItemClickListener<Movie>? = null

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): CustomVH {
        val itemRecycleviewBinding: ItemRecycleviewBinding = DataBindingUtil
            .inflate(LayoutInflater.from(parent.context), R.layout.item_recycleview, parent, false)
        return CustomVH(itemRecycleviewBinding)
    }

    override fun getItemCount(): Int {
        return movies?.size ?: 0
    }

    override fun onBindViewHolder(holder: CustomVH, position: Int) {
        movies?.run {
            if (mOnItemClickListener != null) {
                holder.itemView.setOnClickListener {
                    mOnItemClickListener?.onItemClick(this@run[holder.layoutPosition], it)
                }
            }
            holder.binding.movie = this[position]

            val myImageView = holder.binding.movieCover
            val imageViewHeight = imageViewWidth * Integer.parseInt(this[position].imageOfMovieHeight) /
                    Integer.parseInt(this[position].imageOfMovieWidth)
            val imageViewParams = LinearLayout.LayoutParams(imageViewWidth, imageViewHeight)
            val imagePaddingLeftOrRight = 10
            myImageView.setPadding(imagePaddingLeftOrRight, 10, imagePaddingLeftOrRight, 10)
            myImageView.layoutParams = imageViewParams

            Glide.with(holder.itemView.context)
                .load(this[position].imageOfMovie)
                .apply(fitCenterTransform())
                .into(myImageView)

            holder.binding.executePendingBindings()
        }
    }

    fun setMovies(movies: List<Movie>) {
        this.movies = movies
        notifyDataSetChanged()
    }

    inner class CustomVH(var binding: ItemRecycleviewBinding) : RecyclerView.ViewHolder(binding.root)
}